cmd/rebasehelpers/util/git.go

```diff
@@ -0,0 +1,97 @@
+package util
+
+import (
+	"bytes"
+	"fmt"
+	"os/exec"
+	"strings"
+)
+
+type Commit struct {
+	Sha     string
+	Summary string
+	Files   []File
+}
+
+func (c Commit) DeclaresUpstreamChange() bool {
+	return strings.Contains(strings.ToLower(c.Summary), "upstream")
```

Comment:
We could. In the end we're talking about a heuristic, so we have to find some balance. I can contrive summaries that would break the "starts with" approach and some that break the reservation of "upstream" (just consider something like an "uptream" typo).

I'm find with HasPrefix(upstream) if you guys are.